c7n/resources/cloudfront.py

```diff
@@ -181,6 +183,66 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('check-s3-origin')
+class CheckS3Origin(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront, and verify ownership.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cfront-test
+                resource: distribution
+                filters:
+                - type: check-s3-origin
+                  accounts_from:
+                    url: *accounts-list
+                    expr: accounts-list.canonical_id
+    """
+
+    schema = type_schema(
+        'check-s3-origin',
+        accounts_from=ValuesFrom.schema)
+
+    permissions = ('s3:GetBucketAcl',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def process(self, resources, event=None):
+
+        accounts = ValuesFrom(self.data['accounts_from'], self.manager).get_values()
+        results = []
+
+        client = local_session(self.manager.session_factory).client(
+            's3', region_name=self.manager.config.region)
+
+        for r in resources:
+            for x in r['Origins']['Items']:
+                if 'S3OriginConfig' in x:
+                    target_bucket = x['DomainName'].split('.', 1)[0]
+                    try:
+                        b = client.get_bucket_acl(
+                            Bucket=target_bucket
+                        )
+                        self.log.debug("Target bucket {0} exists.".format(target_bucket))
+                        if accounts and b['Owner']['ID'] not in accounts:
+                            self.log.debug("Bucket {0} owner not in accounts list.".
+                                           format(target_bucket))
+                        else:
+                            r['c7n:s3-origin'] = True
+                            results.append(r)
+                    except ClientError as e:
```

Comment:
the new way for exceptions is to use the autogenerated exception classes off of the client

ie. `except client.exceptions.AccessDenied:`

we might want to be tracking on NoSuchBucket, since that can also be a cross account s3 origin, which defeats the purpose of the accounts from being done here since creds in account A, won't see buckets in Account B